src/lex/cpp.rs

```diff
@@ -1534,4 +1589,12 @@ int main(){}
         ";
         assert_same(original, "a(1)");
     }
+    #[test]
+    // https://github.com/jyn514/rcc/issues/356
+    fn preprocess_only() {
+        assert_same_exact("int \t\n\r     main() {}", "int \t\n\r     main() {}");
+        assert_same_exact("int/* */main() {}", "int main() {}");
+        assert_same_exact("int/*\n\n\n*/main() {}", "int\n\n\nmain() {}");
```

Comment:
Hmm, this behavior looks a little confusing. Is there a reason you kept newlines inside of block comments?

Reply:
This seems to be the behavior of clang. For example,
```
int main() {
 /*




 */
}
```
preprocesses to
```
# 1 "test.c"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 363 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "test.c" 2
int main() {






}
```

Reply:
I may need to look at the documentation though since
```
int main() {/*




*/}
```
mysteriously preprocesses to
```
# 1 "test.c"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 363 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "test.c" 2
int main() { }
```

Reply:
Relevant part of the standard ([5.1.1.2 Translation phases](http://port70.net/~nsz/c/c11/n1570.html#5.1.1.2)):
>  The source file is decomposed into preprocessing tokens7) and sequences of white-space characters (including comments). A source file shall not end in a partial preprocessing token or in a partial comment. Each comment is replaced by one space character. New-line characters are retained. Whether each nonempty sequence of white-space characters other than new-line is retained or replaced by one space character is implementation-defined.

This seems to me to say that clang's behavior in your first example is correct, and the behavior in the second is a bug. For context, gcc always behaves like clang in the second example, i.e. it always deletes newlines in comments:

```c
gcc -x c -E -P -
int main() {
 /*




 */
}
```
preprocesses to
```c
int main() {
}
```
and
```c
int main() {/*




*/}
```
preprocesses to `int main() { }`.

Reply:
tcc behaves the same as gcc.